src/Adapter/Customer/Group/Repository/GroupRepository.php

```diff
@@ -29,19 +29,27 @@
 
 use Group as CustomerGroup;
 use PrestaShop\PrestaShop\Adapter\CoreException;
+use PrestaShop\PrestaShop\Adapter\Customer\Group\Event\GroupManagementEvent;
 use PrestaShop\PrestaShop\Core\Domain\Customer\Group\Exception\CannotAddGroupException;
 use PrestaShop\PrestaShop\Core\Domain\Customer\Group\Exception\CannotDeleteGroupException;
 use PrestaShop\PrestaShop\Core\Domain\Customer\Group\Exception\CannotUpdateGroupException;
 use PrestaShop\PrestaShop\Core\Domain\Customer\Group\Exception\GroupNotFoundException;
 use PrestaShop\PrestaShop\Core\Domain\Customer\Group\ValueObject\GroupId;
 use PrestaShop\PrestaShop\Core\Domain\Shop\ValueObject\ShopId;
 use PrestaShop\PrestaShop\Core\Repository\AbstractMultiShopObjectModelRepository;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
 
 /**
  * Provides methods to access Group data storage
  */
 class GroupRepository extends AbstractMultiShopObjectModelRepository
 {
+    public function __construct(
+        private readonly ?EventDispatcherInterface $eventDispatcher = null,
```

Comment:
I don't think this should be nullable, let's exploit this thing fully Adding new parameters for DI i s accepted by our ADR on backward promises Especially for services like this one that should not be instantiated manually and are only used via the container